src/client/Microsoft.Identity.Client/OAuth2/Throttling/SingletonThrotlingManager.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using Microsoft.Identity.Client.Internal.Requests;
+using Microsoft.Identity.Client.TelemetryCore.Internal.Events;
+using Microsoft.Identity.Client.Utils;
+
+namespace Microsoft.Identity.Client.OAuth2.Throttling
+{
+
+    /// <summary>
+    /// Throttling occurs
+    /// <list type="bullet">
+    /// <item>Afetr receving an RetryAfter header</item>
+    /// <item>After receiving 429, 5xx HTTP status.</item>
+    /// <item>After receiving a UI Interaction required signal</item>
+    /// </list>
+    /// 
+    /// This class manages the throttling providers and is itself a provider
+    /// </summary>
+    /// <remarks>
+    /// Client Throttling spec https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1624
+    /// </remarks>
+    internal class SingletonThrottlingManager : IThrottlingProvider
```

Comment:
name of file is missing a "t"

Reply:
Hah, nice catch. I almost never have typos in my code... :D